x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -510,9 +513,13 @@ const getAgentAndPoliciesForEndpointsList = async (
   // Create an array of unique policy IDs that are not yet known to be non-existing.
   const policyIdsToCheck = Array.from(
     new Set(
-      hosts
-        .filter((host) => !currentNonExistingPolicies[host.metadata.Endpoint.policy.applied.id])
-        .map((host) => host.metadata.Endpoint.policy.applied.id)
+      hosts.reduce((acc: string[], host) => {
```

Comment:
as Set will prevent you having duplicated values, you can just map the results.

```
const policyIdsToCheck = Array.from(new Set( hosts.map(host => host.metadata.Endpoint.policy.applied.id)));

or

const policyIdsToCheck = hosts.reduce((acc: string[], host) => {
        const appliedPolicyId = host.metadata.Endpoint.policy.applied.id;
        if (!currentNonExistingPolicies[appliedPolicyId]) {
          acc.push(appliedPolicyId);
        }
        return acc;
      }, [])
```

Reply:
Ah yes. Now the `Set` is redundant!

Reply:
done 728967bb8d908db0642ed43fd149a75602aee601

Reply:
oh wait! My second solution will have duplicated values in the `acc` array. You should go for the first one using Set:
`const policyIdsToCheck = Array.from(new Set( hosts.map(host => host.metadata.Endpoint.policy.applied.id)));`

Reply:
another solution could be:

```
const policyIdsToCheck = Object.keys(hosts.reduce((acc, host) => {
        const appliedPolicyId = host.metadata.Endpoint.policy.applied.id;
        if (!currentNonExistingPolicies[appliedPolicyId]) {
          acc[appliedPolicyId]: true;
        }
        return acc;
      }, {}))
```

Reply:
I'll stick with the `Set` approach for removing duplicates I find that simpler to reason with.

Reply:
done 728967bb8d908db0642ed43fd149a75602aee601 😄

Reply:
Set works perfectly, next time you can use some lodash utils to avoid creating new Set instance and destructure it right away: https://lodash.com/docs/4.17.15#uniq. But no need to do it now, it works 🙂

Reply:
i.e. using lodash: 
```
const policyIdsToCheck = uniq(reduce(hosts, (acc, host) => {
       const appliedPolicyId = host.metadata.Endpoint.policy.applied.id;
        if (!currentNonExistingPolicies[appliedPolicyId]) {
          acc.push(appliedPolicyId);
        }
        return acc;
      }, []))
```

Reply:
Yes indeed. I like to use `lodash` helpers only in case of deep object comparisons. Since here it's only string ids I think the native `Set` is alright.